src/mon/AuthMonitor.cc

```diff
@@ -1083,6 +1083,34 @@ int AuthMonitor::do_osd_new(
   return 0;
 }
 
+bool AuthMonitor::valid_caps(
+    const string& type,
+    const string& caps,
+    ostream *out)
+{
+  if (type == "mon" || type == "mgr") {
+    MonCap tmp;
+    if (!tmp.parse(caps, out)) {
+      return false;
+    }
+  } else if (type == "osd") {
+    OSDCap ocap;
+    if (!ocap.parse(caps, out)) {
+      return false;
+    }
+  } else if (type == "mds") {
+    MDSAuthCaps mdscap;
+    if (!mdscap.parse(g_ceph_context, caps, out)) {
+      return false;
+    }
+  } else {
+    if (out) {
+      *out << "unknown cap type '" << type << "'";
+    }
```

Comment:
this breaks 
```
2018-04-14T05:51:59.176 INFO:tasks.workunit.client.0.mira022.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:49: expect_false:  ceph auth add client.xx mon 'allow *' osd 'allow *' invalid 'allow *'
2018-04-14T05:51:59.608 INFO:tasks.workunit.client.0.mira022.stderr:unknown cap type 'invalid'
2018-04-14T05:51:59.627 INFO:tasks.workunit.client.0.mira022.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:49: expect_false:  return 1
```

see http://pulpito.ceph.com/kchai-2018-04-14_03:18:44-rados-wip-kefu-testing-2018-04-14-0918-distro-basic-mira/2396543/

Reply:
Right, I suppose the test needs to be amended as per my earlier discussion with @liewegas

Reply:
huh, my bad; the problem is not returning false there